src/Aspire.Hosting.Garnet/GarnetBuilderExtensions.cs

```diff
@@ -0,0 +1,131 @@
+    /// var garnet = builder.AddGarnet("garnet"); 
+    /// api.WithReference(garnet);
+    ///  
+    /// builder.Build().Run(); 
+    /// </code>
+    /// <remarks>Use in Api with Aspire.StackExchange.Redis</remarks>
+    /// <code>
+    /// var builder = WebApplication.CreateBuilder(args);
+    /// builder.AddRedisClient("garnet");
+    ///
+    /// var multiplexer = builder.Services.BuildServiceProvider()
+    ///                                   .GetRequiredService&lt;IConnectionMultiplexer&gt;();
+    /// 
+    /// var db = multiplexer.GetDatabase();
+    /// db.HashSet("key", [new HashEntry("hash", "value")]);
+    /// var value = db.HashGet("key", "hash");
+    /// </code>
+    /// </example>
+    /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/>.</param>
+    /// <param name="name">The name of the resource. This name will be used as the connection string name when referenced in a dependency.</param>
+    /// <param name="port">The host port to bind the underlying container to.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<GarnetResource> AddGarnet(this IDistributedApplicationBuilder builder, string name, int? port = null)
+    {
+        var garnet = new GarnetResource(name);
+        return builder.AddResource(garnet)
+                      .WithEndpoint(port: port, targetPort: 6379, name: GarnetResource.PrimaryEndpointName)
+                      .WithImage(GarnetContainerImageTags.Image, GarnetContainerImageTags.Tag)
+                      .WithImageRegistry(GarnetContainerImageTags.Registry);
+    }
+
+    /// <summary>
+    /// Adds a named volume for the data folder to a Garnet container resource and enables Garnet persistence.
+    /// </summary>
+    /// <remarks>
+    /// Use <see cref="WithPersistence(IResourceBuilder{GarnetResource}, TimeSpan?, long)"/> to adjust Garnet persistence configuration, e.g.:
+    /// <code>
+    /// var cache = builder.AddGarnet("cache")
+    ///                    .WithDataVolume()
+    ///                    .WithPersistence(TimeSpan.FromSeconds(10), 5);
+    /// </code>
+    /// </remarks>
+    /// <param name="builder">The resource builder.</param>
+    /// <param name="name">The name of the volume. Defaults to an auto-generated name based on the application and resource names.</param>
+    /// <param name="isReadOnly">
+    /// A flag that indicates if this is a read-only volume. Setting this to <c>true</c> will disable Garnet persistence.<br/>
+    /// Defaults to <c>false</c>.
+    /// </param>
+    /// <returns>The <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<GarnetResource> WithDataVolume(this IResourceBuilder<GarnetResource> builder, string? name = null, bool isReadOnly = false)
+    {
+        builder.WithVolume(name ?? VolumeNameGenerator.CreateVolumeName(builder, "data"), "/data", isReadOnly);
```

Comment:
usually the target volumes are a bit more specific -- eg is `/data` too generic that might interfere with other potential areas -- usually these are `/var/garnet/data` or something like that -- the docs are sparse, but may consider changing this.

Reply:
done